### PR TITLE
Enhance Login & Registration Pages

### PR DESCRIPTION
**Pull request by McNaBry**
## Description

- Add icons to input fields
- Revamp password requirement checklist
- Adjust height styling for login and registration pages. (Took into account the height of the navigation bar)
- (For nav bar) Add fix for nav bar menu positioning
- (For match page) Add spinner when redirecting to workspace

## Checklist

- [x] I have updated documentation
- [x] All tests passing

## Screenshots (if applicable)
image
image
image
image
image

### CODE REVIEW

File: frontend/src/app/account/register.component.html [hunk 1]
```diff
@@ -12,42 +15,89 @@ <h2 class="mt-0 align-self-start">Register</h2>
...
+                [feedback]="false">
             </p-password>
             @if (isPasswordWeak) {
                 <small class="text-red-300">The provided password is too weak</small>
             } @else if (isPasswordInvalid) {
                 <small class="text-red-300">The provided password contains invalid characters</small>
             }
+            <ul class="pl-0 m-0 text-sm line-height-3">
+                <li class="flex gap-2 align-items-center">
+                    @if (isPasswordLowercase) {
```
Comment (samuelim01) [inline frontend/src/app/account/register.component.html:47]: Could we also hide the criteria until the user starts modifying the password field (i.e. `passwordControl.dirty`), and hide the criteria once all are satisfied (`passwordControl.valid`)?

**Review by samuelim01 — CHANGES_REQUESTED** [review]
LGTM, just some small code quality changes and suggestions 🙏

File: frontend/src/app/account/register.component.ts [hunk 2]
```diff
@@ -70,6 +82,31 @@ export class RegisterComponent {
         return emailControl.dirty && emailControl.invalid;
     }
 
+    get isPasswordLowercase(): boolean {
```
[the remark this answers is not in the document]
Comment (McNaBry) [reply frontend/src/app/account/register.component.ts:85]: Yeah was wondering if I should just use something else. Got any ideas?

File: frontend/src/app/account/register.component.html [hunk 3]
```diff
@@ -12,42 +15,89 @@ <h2 class="mt-0 align-self-start">Register</h2>
...
-                        <li>At least one numeric</li>
-                        <li>At least one special character</li>
-                        <li>Minimum 8 characters</li>
-                    </ul>
-                </ng-template>
+                [toggleMask]="true"
+                [feedback]="false">
             </p-password>
             @if (isPasswordWeak) {
                 <small class="text-red-300">The provided password is too weak</small>
```
[the remark this answers is not in the document]
Comment (McNaBry) [reply frontend/src/app/account/register.component.html:41]: Ok sure was wondering about this too. I think cleaner if removed.

File: frontend/src/app/account/register.component.html (hunk 1 above)
Comment (McNaBry) [reply frontend/src/app/account/register.component.html:47]: Yup good idea. Was thinking whether I should do that haha

File: frontend/src/app/account/register.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (samuel-us2) [reply frontend/src/app/account/register.component.ts:85]: Maybe `passwordHasNoX`?

[the remark this answers is not in the document]
Comment (samuelim01) [reply frontend/src/app/account/register.component.ts:85]: Maybe `passwordHasNoX`?

**Review by samuelim01 — APPROVED** [review]
LGTM

File: frontend/src/app/navigation-bar/navigation-bar.component.ts [hunk 4]
```diff
@@ -28,6 +29,19 @@ export class NavigationBarComponent implements OnInit {
...
+    hideMenuOnScroll() {
+        if (this.menu && this.menu.overlayVisible) {
+            this.menu.hide();
+        }
+
+        // This makes hiding instant
+        const overlay = document.querySelector('.p-menu-overlay');
+        if (overlay) {
+            (overlay as HTMLElement).style.display = 'none';
+        }
```
Comment (samuelim01) [inline frontend/src/app/navigation-bar/navigation-bar.component.ts:42]: @limcaaarl Is this bit needed? I think it's generally not recommended to manipulate the DOM directly.

**Pull request merged**